Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.cpp

```diff
@@ -327,9 +327,13 @@ PlatformWheelEvent RemoteLayerTreeEventDispatcher::filteredWheelEvent(const Plat
 
 RemoteLayerTreeDrawingAreaProxyMac& RemoteLayerTreeEventDispatcher::drawingAreaMac() const
 {
-    auto* drawingArea = dynamicDowncast<RemoteLayerTreeDrawingAreaProxy>(m_scrollingCoordinator->webPageProxy().drawingArea());
-    ASSERT(drawingArea && drawingArea->isRemoteLayerTreeDrawingAreaProxyMac());
-    return *static_cast<RemoteLayerTreeDrawingAreaProxyMac*>(drawingArea);
+    return ([&](auto* drawingArea) -> auto& {
+        auto* drawingAreaProxy = dynamicDowncast<RemoteLayerTreeDrawingAreaProxy>(drawingArea);
+        RELEASE_ASSERT(drawingAreaProxy);
+        auto* macProxy = dynamicDowncast<RemoteLayerTreeDrawingAreaProxyMac>(drawingAreaProxy);
```

Comment:
Why are we using dynamic downcast instead downcast? Clearly we expect the types to match. Also unclear why we cannot cast directly to RemoteLayerTreeDrawingAreaProxyMac

Reply:
That's because isRemoteLayerTreeDrawingAreaProxyMac isn't defined on DrawingAreaProxy. We can move the virtual function there though.